src/platform_impl/windows/window.rs

```diff
@@ -569,41 +575,41 @@ impl Window {
         if let Some(ref window_icon) = window_icon {
             window_icon
                 .inner
-                .set_for_window(self.window.0, IconType::Small);
+                .set_for_window(self.hwnd(), IconType::Small);
         } else {
-            icon::unset_for_window(self.window.0, IconType::Small);
+            icon::unset_for_window(self.hwnd(), IconType::Small);
         }
         self.window_state.lock().window_icon = window_icon;
     }
 
     #[inline]
     pub fn set_enable(&self, enabled: bool) {
-        unsafe { winuser::EnableWindow(self.hwnd() as _, enabled as _) };
+        unsafe { EnableWindow(self.hwnd(), enabled.into()) };
     }
 
     #[inline]
     pub fn set_taskbar_icon(&self, taskbar_icon: Option<Icon>) {
         if let Some(ref taskbar_icon) = taskbar_icon {
             taskbar_icon
                 .inner
-                .set_for_window(self.window.0, IconType::Big);
+                .set_for_window(self.hwnd(), IconType::Big);
         } else {
-            icon::unset_for_window(self.window.0, IconType::Big);
+            icon::unset_for_window(self.hwnd(), IconType::Big);
         }
         self.window_state.lock().taskbar_icon = taskbar_icon;
     }
 
     pub(crate) fn set_ime_position_physical(&self, x: i32, y: i32) {
-        if unsafe { winuser::GetSystemMetrics(winuser::SM_IMMENABLED) } != 0 {
-            let mut composition_form = COMPOSITIONFORM {
+        if unsafe { GetSystemMetrics(SM_IMMENABLED) } != 0 {
+            let composition_form = COMPOSITIONFORM {
                 dwStyle: CFS_POINT,
                 ptCurrentPos: POINT { x, y },
                 rcArea: unsafe { mem::zeroed() },
             };
             unsafe {
-                let himc = winapi::um::imm::ImmGetContext(self.window.0);
-                winapi::um::imm::ImmSetCompositionWindow(himc, &mut composition_form);
-                winapi::um::imm::ImmReleaseContext(self.window.0, himc);
+                let himc = ImmGetContext(self.hwnd());
+                ImmSetCompositionWindow(himc, &composition_form);
```

Comment:
This function is documented [on MSDN](https://docs.microsoft.com/en-us/windows/win32/api/imm/nf-imm-immsetcompositionwindow) as taking an `LPCOMPOSITIONFORM` (`*mut COMPOSITIONFORM`), but `windows-sys` says it takes a `*const COMPOSITIONFORM`.

Reply:
Probably some bug in the metadata, which I can't fix...

Reply:
The metadata (and SAL) says the parameter is input (only) so `*const` is applied as it is not expected to be mutable.